syntax = "proto3";

package TC_Proto;

import "map_renderer.proto"; 
import "transport_router.proto";

message StopInfo {
	string stop_name = 1;
	repeated uint32 bus_names_indexes = 2;
	double lat = 3;
    double lng = 4;
	uint32 index = 5;
}

message BusInfo {
	string bus_name = 1;
	uint32 stops_on_route = 7;
	repeated uint32 stops_indexes = 2;
	uint32 unique_stops = 3;
	uint64 route_length_on_road = 4;
	double curvature = 5;
	bool is_ring = 6;
}

message TransportCatalogue {
	repeated StopInfo stops_info = 1;
	repeated BusInfo buses_info = 2;
	MR_Proto.RenderSettings render_settings = 3;
	Router_proto.RoutesData router = 4;
}

// C:/dev/ProtoBuf/path/to/protobuf/package/bin/protoc --cpp_out . transport_catalogue.proto